packages/d2-app/src/commands/scripts/release.js

```diff
@@ -0,0 +1,82 @@
+const { reporter } = require('@dhis2/cli-helpers-engine')
+
+const semanticRelease = require('semantic-release')
+
+const options = {
+    branch: 'master',
+    version: 'v${version}',
+
+    plugins: [
+        '@semantic-release/commit-analyzer',
+        '@semantic-release/release-notes-generator',
+
+        [
+            '@semantic-release/changelog',
+            {
+                changelogFile: 'CHANGELOG.md',
+            },
+        ],
+
+        '@semantic-release/npm',
+
+        [
+            '@semantic-release/git',
+            {
+                assets: ['CHANGELOG.md', 'package.json', 'yarn.lock'],
+                message:
+                    'chore(release): cut ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}',
+            },
+        ],
+
+        '@semantic-release/github',
+    ],
+}
+
+const handler = async ({ name }) => {
+    const config = {
+        env: {
+            ...process.env,
+            GIT_AUTHOR_NAME: '@dhis2-bot',
+            GIT_AUTHOR_EMAIL: 'ci@dhis2.org',
+            GIT_COMMITTER_NAME: '@dhis2-bot',
+            GIT_COMMITTER_EMAIL: 'ci@dhis2.org',
+        },
+    }
+
+    try {
+        const result = await semanticRelease(options, config)
```

Comment:
We should add a `--dry-run` CLI option which passes through to the same option for semantic release, so we can test release logic locally

Reply:
I can do that. Right now you are barred from doing a local release as it checks for a CI environment by default. I'll make that explicit too.

Reply:
Introduced `--no-dry-run`/`--dry-run` and `--no-ci`/`--ci`

Reply:
Do you think we could get away with not allowing `--ci`?  Maybe we need it

Reply:
We can get away with not having `--dry-run` and `--ci`, I think.

`--ci` defaults to `true` in semantic-release, so the default is to only allow CI environments to run the release.

So if I run the release command just the defaults it automatically goes into dry run mode.

```
d2 app scripts release
...
[1:59:08 PM] [semantic-release] › ⚠  This run was not triggered in a known CI environment, running in dry-run mode.
```

To actually run the `release` locally I need to pass in `--no-ci`.

Reply:
>Also, is this actually for apps or just for libraries? Apps still have a different (non-npm) release process right?

Yeah, I'm pondering adding a `--mode [app/lib]` switch, as the only thing we would need to _not_ do for apps is publish to NPM:  https://github.com/dhis2/cli/blob/86c2edd83d92d72f08115955c0e0b377c2591d4c/packages/d2-app/src/commands/scripts/release.js#L24

Reply:
What about a `--publish` flag, set for libs but not apps?  Or check the `publishConfig` in package.json?

( GitHub hates email )

Reply:
> So if I run the release command with the defaults it automatically goes into dry run mode.

Cool, yeah no need for dry-run then!  Sorry I didn't realize you could preview it locally without

Reply:
Sure. :+1:

An alternative is just setting `"private": true"` in `package.json` for Apps (not verified).

Reply:
I would like to have a `--publish [target]` in that case, so we can publish to NPM for libs and App Store for apps.

Reply:
0c6fd99

Reply:
This solution looks good to me!